### DISCUSSION TITLE
Upgrading http to ws

### DISCUSSION THREAD

**Discussion by ghost** [uNetworking/uWebSockets.js#141] [Q&A] [answered]
Does uWebsocket.js supports upgrading from http socket to websocket **manually**?

**Comment by ghost** [comment]
No, why?

**Comment by ghost** [comment]
I'd like to use uWebsockets.js as replacement for ws module in engine.io and that feature would be handy for it.

**Comment by ghost** [comment]
Nope

**Comment by Sceat** [comment]
It is actually useful to deny clients before upgrading (ex: auth)

**Comment by ghost** [comment]
Yes that is a feature being worked on. But in any case you shouldn't use Engine.io it makes no sense

**Comment by Sceat** [comment]
Should this issue be reopened ? for the Upgrading part 
*(as indeed engine.io stuff is totally unrelated)*

I would actually use uWs when this land

**Comment by ghost** [comment]
There is an open issue in the main repo

**Comment by ghost** [comment, marked as answer]
UpgradeAsync and UpgradeSync are examples of doing auth
